tests/utils/test_optional_import.py

```
@@ -75,6 +77,34 @@ def versioning(module, ver, a):
+            return ref
+
+        ref = _do_import()
+        gc.collect()
+        self.assertIsNone(ref(), "optional_import is leaking frame references via traceback")
+
+    def test_failed_import_shows_traceback_string(self):
+        """Verify the error message includes the original traceback as a string."""
+        mod, flag = optional_import("nonexistent_module_for_tb_test")
+        self.assertFalse(flag)
+        with self.assertRaises(OptionalImportError) as ctx:
+            mod.something
```

Ruff (0.15.9)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)